### DISCUSSION TITLE
zkSync Sepolia RPC failing non-deterministically

### DISCUSSION THREAD

**Discussion by zjesko** [zkSync-Community-Hub/zksync-developers#469] [General]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

Trying to do normal function calls on zksync sepolia testnet

### What is the specific issue or error you're encountering?

https://chainlist.org/chain/300

All of the above mentioned RPCs fail all at once sometimes and starts working again. This happens on interacting with not only our protocol but other protocols on testnet as well

### Can you share the error messages or logs you're receiving, if any?

cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0x97B1804438321cAf638E2D2aE9385dd4759D9C77","to":"0xF8C4B3D6C227A8fed4bB263e7D9658Dbe1Ba4c18","value":{"type":"BigNumber","hex":"0x01f438daa06000"},"data":"0xac9650d800000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000c000000000000000000000000000000000000000000000000000000000000000447d39aaf1000000000000000000000000f4629fdfde544c4f58120faa055b73603974d58c0000000000000000000000000000000000000000000000000001f438daa060000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002a44a393a41000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000

### Have you made any recent changes to the contract before encountering this issue?

No, no changes were made to the contract, it sometimes works and sometimes fails unexpectedly

### Are there any external libraries or contracts that your contract interacts with?

pyth and open zeppelin

### Can you provide the relevant portions of your contract code where the issue is occurring?

All portions of code

### Have you tried to isolate the problem, and if so, what were the results?

Happening across all RPCs and all protocols but resumes working after a while

### What steps have you already taken to try to resolve the issue?

none

**Comment by bxpana** [comment]
Thanks for reporting. Will look into this

**Reply by bxpana** [reply]
@zjesko can you share a link to the repo or share the code so we can check this out. How often is this happening?
